data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep

```diff
@@ -110,10 +110,14 @@
         <bridge>yes</bridge>
         <bridge_forwarddelay>15</bridge_forwarddelay>
         <!-- Use specified interface instead of a list to avoid ip lease inconsistency. Please refer to poo#135641. -->
-        % if ($check_var->('SUT_NETDEVICE', 'eth1') or $check_var->('SUT_NETDEVICE', 'em1')) {
+	% if ($get_var->('SUT_NETDEVICE')) {
+	  % if ($get_var->('SUT_NETDEVICE') =~ /((?:[0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})/) {
```

Comment:
I am curious about this pattern ```/((?:[0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})/```. What are you trying to match >?

Reply:
the MAC of ipv4 or ipv6. 
```
~/gitcode/salt-pillars-openqa/openqa> grep SUT_NETDEVICE workerconf.sls 
        SUT_NETDEVICE: ec:2a:72:02:84:20
        SUT_NETDEVICE: eth1
        SUT_NETDEVICE: F4:02:70:B9:3A:66
```

Reply:
add inline comment as well.

Reply:
Why not just give the real device name to ```SUT_NETDEVICE``` ? So you can put value of it in ```bridge_ports``` directly. I do not think it makes any sense to differentiate the device name based on whether ```SUT_NETDEVICE``` is given mac address or real device name. It should always has device name.

Reply:
because `SUT_NETDEVICE` is used elsewhere, fg. the bootscript to install the host, 
```
kernel http://openqa.qa2.suse.asia/assets/repo/fixed/SLE-12-SP5-Server-DVD-x86_64-GM-DVD1/boot/x86_64/loader/linux install=http://openqa.qa2.suse.asia/assets/repo/fixed/SLE-12-SP5-Server-DVD-x86_64-GM-DVD1?device=B8:CB:29:DB:C1:7A ifcfg=B8:CB:29:DB:C1:7A=dhcp4   root=/dev/ram0 initrd=initrd  textmode=1
```

Reply:
Anyway there is no logic or proper relation between "mac being used" and "em1 interface". I do not think you should use it this way.

Reply:
It has been being used this way in test scripts. I don't know which tests or which teams are using it. My PR is the quickest way to fix the bonding issue to make our tests workable meanwhile not make other tests broken.

Reply:
@Julie-CAO @waynechen55   To me, https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19726/files#r1677298216 is valid point. SUT_NETDEVICE may not always be em1, although maybe now it is the value. 
On the other hand, I understand that it is not easy to detect the real name of the SUT_NETDEVICE at prepare_profile stage if it is given mac address. 

To find an easy solution, how do you think about introducing one more variable to specify the bridge port for our dev_host_xx.xml.ep?

Reply:
I think introduce a new setting ```VIRT_HOST_BRIDGE_PORT``` is better. It clearly specify its usage and avoid ambiguity.

Reply:
I agree that introducing a new setting for sure is the perfect solution, if you all think it is no hurry to get the issue fixed, and we have only sle15sp7 one release need this setting. There are what we need to do:

- submit a MR to add the new setting to workers.ini
- adapt our automation tests with these two settings. I guest a few scripts need to modify, ipxe_install.pm, boot_from_pxe.pm, 2 autoyast templates, one or two other feature tests scripts.
- an entire verification tests

Based on these changes above, if the time is ok, I'd like to know which task should I do firstly.

- this PR
- a bug follow up https://bugzilla.suse.com/show_bug.cgi?id=1221812#c19
- vgpu automation
- fix bare-metal 3 boot order

Reply:
It seems more work than expected :(. Sorry for overlooking some details. @Julie-CAO  How long do you expect it to need?

I do not quite follow details about that infra ticket related with this PR. Is it correct that without the fix, we will impact other team's infra/automation? If yes, I think fixing this is first priority. It's fine to merge it now, then go directly with enhancement.

Reply:
> It seems more work than expected :(. Sorry for overlooking some details. @Julie-CAO How long do you expect it to need?

It will cost a few days(also considering OSD is not fuctioning well recently).

> I do not quite follow details about that infra ticket related with this PR. Is it correct that without the fix, we will impact other team's infra/automation? If yes, I think fixing this is first priority. It's fine to merge it now, then go directly with enhancement.

No, if we don't run any sle12sp5 autoyast installation on those 3 machines I listed in slack, we will not impact the subnet or other teams. The impacts to us: 

- we can't run sle12sp5 host installation tests. 
- bare-metal2 is disconnected.

Reply:
You can keep ```SUT_NETDEVICE``` anywhere else. Just introducing a new setting for new usage. I do not think it is good to merge this without this change. I do not think there is any urgent requirement to merge this pull request at the moment.

Reply:
> You can keep `SUT_NETDEVICE` anywhere else. Just introducing a new setting for new usage.

No, this setting is used both as interface name and MAC in test scripts. We need split them completely since we will introduce a new setting.

Reply:
> > You can keep `SUT_NETDEVICE` anywhere else. Just introducing a new setting for new usage.
> 
> No, this setting is used both as interface name and MAC in test scripts. We need split them completely since we will introduce a new setting.

I still do not think you should rush with compromised quality. If this is very urgent, the make this the highest priority task. And as I said, also good opportunity of refining relevant settings for machines.

Reply:
> > You can keep `SUT_NETDEVICE` anywhere else. Just introducing a new setting for new usage.
> 
> No, this setting is used both as interface name and MAC in test scripts. We need split them completely since we will introduce a new setting.

@Julie-CAO 
Not sure if we are on the same page... I think, in the final solution, we can keep `SUT_NETDEVICE` untouched. So wherever using it, keeping it the way they are. For the newly introduced setting, use it for port purpose only in the autoyast. It is just that, the value assigners/users, shall make them matched. That is the SUT_NETDEVICE(name or mac) corresponds to the NIC name given in port.

Given the impact listed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19726#discussion_r1678697260, I think it is worthwhile to fix this first. I am fine with doing it with two steps if you want.

Reply:
so, do you mean keep `SUT_NETDEVICE` the current value(some are NIC names and some are ports) and add `BRIDGE_PORTS` to these 3 machines? I am ok with that.

```
~/gitcode/salt-pillars-openqa/openqa> grep SUT_NETDEVICE workerconf.sls 
        SUT_NETDEVICE: ec:2a:72:02:84:20
        SUT_NETDEVICE: eth1
        SUT_NETDEVICE: F4:02:70:B9:3A:66
```

will be changed to be:

```
~/gitcode/salt-pillars-openqa/openqa> grep -e SUT_NETDEVICE -e BRIDGE_PORTS workerconf.sls 
        SUT_NETDEVICE: ec:2a:72:02:84:20
        BRIDGE_PORTS: em1

        SUT_NETDEVICE: eth1
        BRIDGE_PORTS: eth1

        SUT_NETDEVICE: F4:02:70:B9:3A:66
        BRIDGE_PORTS: em1
```

Reply:
I prefer "without s".

Reply:
> I prefer "without s". This is very trivial.

Edited in my previous comments. 

After Alice confirm, I'll begin with the change, and annouce this new setting in team for some machines in BJ after the PR is merged.

Reply:
I am ok with any of your solutions but If it will take much time, I need reset the priority of my tasks in hand.

Reply:
@Julie-CAO Yes.

Reply:
Ok